contracts/swaps/connectors/SwapsImplstETH_ETH.sol

```diff
@@ -0,0 +1,175 @@
+            );
+            if (srcToken == WETH) {
+                amountOut = IwstETH(WSTETH).getWstETHBystETH(amountIn);
+            } else {
+                amountOut = IwstETH(WSTETH).getStETHByWstETH(amountIn);
+            }
+        }
+    }
+
+    function dexAmountOutFormatted(bytes memory payload, uint256 amountIn)
+        public
+        returns (uint256 amountOut, address midToken)
+    {
+        return dexAmountOut(payload, amountIn);
+    }
+
+    function dexAmountIn(bytes memory route, uint256 amountOut)
+        public
+        returns (uint256 amountIn, address midToken)
+    {
+        revert("unsupported");
+    }
+
+    function dexAmountInFormatted(bytes memory payload, uint256 amountOut)
+        public
+        returns (uint256 amountIn, address midToken)
+    {
+        revert("unsupported");
+    }
+
+    function setSwapApprovals(
+        address[] memory /*tokens*/
+    ) public {
+        IERC20(STETH).safeApprove(WSTETH, 0);
+        IERC20(STETH).safeApprove(WSTETH, uint256(-1));
+        IERC20(STETH).safeApprove(address(STETHPOOL), 0);
+        IERC20(STETH).safeApprove(address(STETHPOOL), uint256(-1));
+    }
+
+    function revokeApprovals(
+        address[] memory /*tokens*/
+    ) public {
+        IERC20(STETH).safeApprove(WSTETH, 0);
+        IERC20(STETH).safeApprove(address(STETHPOOL), 0);
+    }
+
+    function _swap(
+        address sourceTokenAddress,
+        address destTokenAddress,
+        address receiverAddress,
+        uint256 minSourceTokenAmount,
+        uint256, /*maxSourceTokenAmount*/
+        uint256 requiredDestTokenAmount,
+        bytes memory payload
+    )
+        internal
+        returns (uint256 sourceTokenAmountUsed, uint256 destTokenAmountReceived)
+    {
+        require(
+            requiredDestTokenAmount == 0,
+            "required dest token amount unsupported"
+        );
+        sourceTokenAmountUsed = minSourceTokenAmount;
+        if (sourceTokenAddress == WETH) {
+            IWeth(WETH).withdraw(minSourceTokenAmount);
+            STETH.call.value(minSourceTokenAmount)("");
+            destTokenAmountReceived = IwstETH(WSTETH).wrap(
+                minSourceTokenAmount
+            );
+        } else {
+            destTokenAmountReceived = IwstETH(WSTETH).unwrap(
+                minSourceTokenAmount
+            );
+            destTokenAmountReceived = STETHPOOL.exchange(
+                1,
+                0,
+                destTokenAmountReceived,
+                abi.decode(payload, (uint256))
+            );
+            IWeth(WETH).deposit.value(destTokenAmountReceived)();
+        }
+        if (receiverAddress != address(this)) {
+            IERC20(destTokenAddress).transfer(
+                receiverAddress,
+                destTokenAmountReceived
+            );
+        }
+    }
+
+    //will be removed on live deployment as unneeded. only used for testing
+    function() external payable {
```

Comment:
why we need it for testing?

Reply:
.transfer needs a fallback to receive ETH to. Our protocol contract handles this already but since test case is isolated, it needs a handling of ETH.

Reply:
ah thats right